### PR TITLE
Increase rabbitmq limits

### PR DESCRIPTION
**Pull request by angstwad**

### CODE REVIEW

File: roles/rabbitmq/handlers/main.yml [hunk 1]
```diff
@@ -1,2 +1,8 @@
-  - name: restart rabbitmq
-    action: service name=rabbitmq-server state=restarted enabled=yes
+---
+- name: restart rabbitmq
+  action: service name=rabbitmq-server state=restarted enabled=yes
+
+- name: login as rabbitmq to update limits
+  command: sudo -i -u rabbitmq
+  register: rabbitmq_login
+  failed_when: rabbitmq_login.rc != 1
```
Comment (j2sol) [inline roles/rabbitmq/handlers/main.yml]: What does this accomplish?

File: roles/rabbitmq/templates/etc/security/limits.d/10-rabbitmq.conf [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# {{ ansible_managed }}
+rabbitmq hard nofile 2048
+rabbitmq soft nofile 2048
```
Comment (j2sol) [inline roles/rabbitmq/templates/etc/security/limits.d/10-rabbitmq.conf]: How do you feel about making these limits a variable? Makes it easier to change, given all 3 are the same.

File: roles/rabbitmq/handlers/main.yml (hunk 1 above)
Comment (angstwad) [inline roles/rabbitmq/handlers/main.yml]: Avoids a reboot to pick up `/etc/security/limits.conf` and `limits.d` changes.  Changes are picked up at login; sudo processes a login with `-i`; shell is `/bin/false` which returns a 1

File: roles/rabbitmq/handlers/main.yml [hunk 3]
```diff
@@ -1,2 +1,8 @@
-  - name: restart rabbitmq
-    action: service name=rabbitmq-server state=restarted enabled=yes
+---
+- name: restart rabbitmq
+  action: service name=rabbitmq-server state=restarted enabled=yes
```
Comment (j2sol) [inline roles/rabbitmq/handlers/main.yml]: While you're changing the file, might as well fix this to remove the `action` bit.

**Comment by j2sol** [comment]
This is good, no need to do another round of CI. If tests pass, will merge.

File: envs/example/defaults-2.0.yml [hunk 4]
```diff
@@ -39,6 +39,7 @@ rabbitmq:
   port: 5672
   user: openstack
   cluster: false
+  nofile: 10240
```
Comment (j2sol) [inline envs/example/defaults-2.0.yml]: Ah, instead of adding it as a defaults-2.0.yml thing, we should make it a role default.

**Comment by j2sol** [comment]
I take this back. The variable should be a role default, and then we can override it in defaults if/when we need to. In general, we should always have a role default to back a role variable.

File: roles/rabbitmq/handlers/main.yml (hunk 3 above)
Comment (angstwad) [inline roles/rabbitmq/handlers/main.yml]: can do.

File: envs/example/defaults-2.0.yml (hunk 4 above)
Comment (angstwad) [inline envs/example/defaults-2.0.yml]: k.

**Comment by angstwad** [comment]
Role default provided.

**Comment by angstwad** [comment]
Tests failed due to a connection error of some kind last time; waiting for tests to pass.  @j2sol, ready for re-review.

**Comment by j2sol** [comment]
retest

**Pull request merged**